env0/resource_project_budget_test.go

```diff
@@ -0,0 +1,163 @@
+						"thresholds": projectBudget.Thresholds,
+					}),
+					Check: resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr(accessor, "project_id", projectBudget.ProjectId),
+						resource.TestCheckResourceAttr(accessor, "amount", strconv.Itoa(projectBudget.Amount)),
+						resource.TestCheckResourceAttr(accessor, "timeframe", projectBudget.Timeframe),
+						resource.TestCheckResourceAttr(accessor, "thresholds.0", strconv.Itoa(projectBudget.Thresholds[0])),
+					),
+				},
+				{
+					Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
+						"project_id": updatedProjectBudget.ProjectId,
+						"amount":     strconv.Itoa(updatedProjectBudget.Amount),
+						"timeframe":  updatedProjectBudget.Timeframe,
+						"thresholds": updatedProjectBudget.Thresholds,
+					}),
+					Check: resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr(accessor, "project_id", updatedProjectBudget.ProjectId),
+						resource.TestCheckResourceAttr(accessor, "amount", strconv.Itoa(updatedProjectBudget.Amount)),
+						resource.TestCheckResourceAttr(accessor, "timeframe", updatedProjectBudget.Timeframe),
+						resource.TestCheckResourceAttr(accessor, "thresholds.0", strconv.Itoa(updatedProjectBudget.Thresholds[0])),
+					),
+				},
+			},
+		}
+
+		runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {
+			gomock.InOrder(
+				mock.EXPECT().ProjectBudgetUpdate(projectBudget.ProjectId, &client.ProjectBudgetUpdatePayload{
+					Amount:     projectBudget.Amount,
+					Timeframe:  projectBudget.Timeframe,
+					Thresholds: projectBudget.Thresholds,
+				}).Times(1).Return(projectBudget, nil),
+				mock.EXPECT().ProjectBudget(projectBudget.ProjectId).Times(2).Return(projectBudget, nil),
+				mock.EXPECT().ProjectBudgetUpdate(updatedProjectBudget.ProjectId, &client.ProjectBudgetUpdatePayload{
+					Amount:     updatedProjectBudget.Amount,
+					Timeframe:  updatedProjectBudget.Timeframe,
+					Thresholds: updatedProjectBudget.Thresholds,
+				}).Times(1).Return(updatedProjectBudget, nil),
+				mock.EXPECT().ProjectBudget(updatedProjectBudget.ProjectId).Times(1).Return(updatedProjectBudget, nil),
+				mock.EXPECT().ProjectBudgetDelete(projectBudget.ProjectId).Times(1).Return(nil),
+			)
+		})
+	})
+
+	t.Run("Create Failure - invalid timeframe", func(t *testing.T) {
+		testCase := resource.TestCase{
+			Steps: []resource.TestStep{
+				{
+					Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
+						"project_id": projectBudget.ProjectId,
+						"amount":     strconv.Itoa(projectBudget.Amount),
+						"timeframe":  "invalid",
+						"thresholds": projectBudget.Thresholds,
+					}),
+					ExpectError: regexp.MustCompile("must be one of: WEEKLY, MONTHLY, QUARTERLY, YEARLY"),
+				},
+			},
+		}
+
+		runUnitTest(t, testCase, func(mock *client.MockApiClientInterface) {})
+	})
+
+	t.Run("Detect drift", func(t *testing.T) {
```

Comment:
how does it detect a drift? I didn't understand this test

Reply:
`mock.EXPECT().ProjectBudget(projectBudget.ProjectId).Times(1).Return(nil, &client.NotFoundError{}),`

It returns a not found error.